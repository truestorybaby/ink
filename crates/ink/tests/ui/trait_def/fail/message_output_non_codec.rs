pub struct NonCodec;

#[ink::trait_definition]
pub trait TraitDefinition {
    #[ink(message)]
    fn message(&self) -> NonCodec;
}

fn main() {}
