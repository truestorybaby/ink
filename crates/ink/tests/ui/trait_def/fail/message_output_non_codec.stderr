error[E0277]: the trait bound `NonCodec: WrapperTypeEncode` is not satisfied
 --> tests/ui/trait_def/fail/message_output_non_codec.rs:6:26
  |
6 |     fn message(&self) -> NonCodec;
  |                          ^^^^^^^^ the trait `WrapperTypeEncode` is not implemented for `NonCodec`
  |
  = help: the following other types implement trait `WrapperTypeEncode`:
            &T
            &mut T
            Arc<T>
            Box<T>
            Cow<'a, T>
            Rc<T>
            String
            Vec<T>
            parity_scale_codec::Ref<'a, T, U>
  = note: required for `NonCodec` to implement `Encode`
note: required by a bound in `DispatchOutput`
 --> src/codegen/dispatch/type_check.rs
  |
  |     T: scale::Encode + 'static;
  |        ^^^^^^^^^^^^^ required by this bound in `DispatchOutput`

error[E0599]: the method `try_invoke` exists for struct `CallBuilder<E, Set<Call<E>>, Set<ExecutionInput<ArgumentList<ArgumentListEnd, ArgumentListEnd>>>, Set<ReturnType<NonCodec>>>`, but its trait bounds were not satisfied
 --> tests/ui/trait_def/fail/message_output_non_codec.rs:5:5
  |
1 | pub struct NonCodec;
  | ------------------- doesn't satisfy `NonCodec: parity_scale_codec::Decode`
...
5 |     #[ink(message)]
  |     ^ method cannot be called due to unsatisfied trait bounds
  |
  = note: the following trait bounds were not satisfied:
          `NonCodec: parity_scale_codec::Decode`
note: the trait `parity_scale_codec::Decode` must be implemented
 --> $CARGO/parity-scale-codec-3.4.0/src/codec.rs
  |
  | pub trait Decode: Sized {
  | ^^^^^^^^^^^^^^^^^^^^^^^
